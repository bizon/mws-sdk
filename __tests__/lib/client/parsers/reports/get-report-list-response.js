const parseXml = require('../../../../../lib/client/parsers')
const parseGetReportListResponse = require('../../../../../lib/client/parsers/reports/get-report-list-response')

describe('lib.client.parsers.reports.get-report-list-response', () => {
  it('should parse the GetReportListResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetReportListResponse xmlns="http://mws.amazonaws.com/doc/2009-01-01/">
        <GetReportListResult>
          <NextToken>2YgYW55IPQhvu5hbCBwbGVhc3VyZS4=</NextToken>
          <HasNext>true</HasNext>
          <ReportInfo>
            <ReportId>898899473</ReportId>
            <ReportType>_GET_MERCHANT_LISTINGS_DATA_</ReportType>
            <ReportRequestId>2278662938</ReportRequestId>
            <AvailableDate>2009-02-10T09:22:33+00:00</AvailableDate>
            <Acknowledged>false</Acknowledged>
          </ReportInfo>
        </GetReportListResult>
        <ResponseMetadata>
          <RequestId>fbf677c1-dcee-4110-bc88-2ba3702e331b</RequestId>
        </ResponseMetadata>
      </GetReportListResponse>`,
    )

    const res = parseGetReportListResponse(
      '/reports:GetReportListResponse',
      doc,
    )

    expect(res).toMatchSnapshot()
  })

  it('should parse the GetReportListByNextTokenResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetReportListByNextTokenResponse xmlns="http://mws.amazonaws.com/doc/2009-01-01/">
        <GetReportListByNextTokenResult>
          <NextToken>none</NextToken>
          <HasNext>false</HasNext>
          <ReportInfo>
            <ReportId>898899473</ReportId>
            <ReportType>_GET_MERCHANT_LISTINGS_DATA_</ReportType>
            <ReportRequestId>2278662938</ReportRequestId>
            <AvailableDate>2009-02-10T09:22:33+00:00</AvailableDate>
            <Acknowledged>false</Acknowledged>
          </ReportInfo>
        </GetReportListByNextTokenResult>
        <ResponseMetadata>
          <RequestId>fbf677c1-dcee-4110-bc88-2ba3702e331b</RequestId>
        </ResponseMetadata>
      </GetReportListByNextTokenResponse>`,
    )

    const res = parseGetReportListResponse(
      '/reports:GetReportListByNextTokenResponse',
      doc,
      true,
    )

    expect(res).toMatchSnapshot()
  })
})
