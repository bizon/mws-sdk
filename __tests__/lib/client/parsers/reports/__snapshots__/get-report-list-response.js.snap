// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.reports.get-report-list-response should parse the GetReportListByNextTokenResponse example response from MWS doc 1`] = `
{
  "getReportListResult": {
    "hasNext": null,
    "nextToken": "none",
    "reportInfos": [
      {
        "acknowledged": false,
        "acknowledgedDate": null,
        "availableDate": 2009-02-10T09:22:33.000Z,
        "reportId": "898899473",
        "reportRequestId": "2278662938",
        "reportType": "_GET_MERCHANT_LISTINGS_DATA_",
      },
    ],
  },
  "responseMetadata": {
    "requestId": "fbf677c1-dcee-4110-bc88-2ba3702e331b",
  },
}
`;

exports[`lib.client.parsers.reports.get-report-list-response should parse the GetReportListResponse example response from MWS doc 1`] = `
{
  "getReportListResult": {
    "hasNext": null,
    "nextToken": "2YgYW55IPQhvu5hbCBwbGVhc3VyZS4=",
    "reportInfos": [
      {
        "acknowledged": false,
        "acknowledgedDate": null,
        "availableDate": 2009-02-10T09:22:33.000Z,
        "reportId": "898899473",
        "reportRequestId": "2278662938",
        "reportType": "_GET_MERCHANT_LISTINGS_DATA_",
      },
    ],
  },
  "responseMetadata": {
    "requestId": "fbf677c1-dcee-4110-bc88-2ba3702e331b",
  },
}
`;
